Python/pywarpx/_libwarpx.py

```diff
@@ -243,6 +243,16 @@ def get_nattr():
     # --- The -3 is because the comps include the velocites
     return libwarpx.warpx_nComps() - 3
 
+def get_nattr_species(species_name):
+    '''
+
+    Get the number of extra real attributes for the given species.
+
+    '''
+    # --- The -3 is because the comps include the velocites
+    return libwarpx.warpx_nCompsSpecies(
+        ctypes.c_char_p(species_name.encode('utf-8'))) - 3
```

Comment:
I am confused by this. The extra arguments the number of arguments in `pc->NumRuntimeRealComps()`, e.g.,  #2126

Reply:
I added this in order to have an easy way to find how many attributes should be given to add_particles - there was a check before that the total number of attributes given are less than or equal to the `NumRealComps() - 3` for the species. This mixing of `NumRuntimeRealComps()` and `NumRealComps()` is because the AddNParticles function effectively treats the weight as an extra attribute (passed to it as part of `attr`) although it isn't a runtime component. So basically to check that `ncomps` is appropriate we can either do `NumRealComps() - 3` or `NumRuntimeRealComps() + 1`.
I've opted for the former.

Reply:
I see, thanks for the clarification. Yes, I see now how it comes from existing code.
I realized the existing wording here (extra attributes) vs. the C++ part (compile-time attributes in SoA + runtime attributes as SoA) is a bit different, maybe we can unify that a bit in a follow-up PR to avoid confusion.